src/Dianoga/ExtensionBasedImageOptimizer.cs

```diff
@@ -17,5 +17,15 @@ public virtual bool CanOptimize(MediaStream stream)
 		}
 
 		public abstract IOptimizerResult Optimize(MediaStream stream);
+
+		protected IOptimizerResult OptimizationSuccessful(IOptimizerResult result)
+		{
+			if (result.SizeBefore < result.SizeAfter)
+			{
+				result.Success = false;
+				result.ErrorMessage = "the optimized image resulted in a larger file size. Using the original instead.";
+			}
+			return result;
+		}
```

Comment:
There's no point in serving back an image if it is larger in size than the original (it happens!). Check the before and after size to ensure optimization occurred.